### PR TITLE
Learning rate scheduler

### PR DESCRIPTION
**Pull request by ss8319**
Added a LR scheduler as discussed with Arnab
Surprisingly neuroprobe uses a fixed lr

### CODE REVIEW

**Comment by ss8319** [comment]
Hi @pritamarnab 
Implemented the LR scheduler here, there is a sample config file too for hyperparameters.
it should be quite easy to replace/add new lr schedulers too

**Comment by ss8319** [comment]
Hey @pritamarnab 
Sorry I missed the previous meeting, just wondering if this helped at all?

**Comment by zparis99** [comment]
Are we ready to merge this? I'm fine with doing so if it is!

**Comment by ss8319** [comment]
interesting do u know why LR scheduler dont help?
Perhaps with LR, u might need to swap hyperparams to see. 
Also with the way the scheduler is wrote it should be v easy to change schedulers-another thing we can try.
This is useful [PyTorch docs.](https://docs.pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.CosineAnnealingLR.html)
